application/src/main/java/org/thingsboard/server/service/state/DefaultDeviceStateService.java

```diff
@@ -651,13 +655,24 @@ private DeviceStateData toDeviceStateData(EntityData ed, DeviceIdInfo deviceIdIn
         TbMsgMetaData md = new TbMsgMetaData();
         md.putValue("deviceName", getEntryValue(ed, EntityKeyType.ENTITY_FIELD, "name", ""));
         md.putValue("deviceType", getEntryValue(ed, EntityKeyType.ENTITY_FIELD, "type", ""));
-        return DeviceStateData.builder()
+        DeviceStateData deviceStateData = DeviceStateData.builder()
                 .customerId(deviceIdInfo.getCustomerId())
                 .tenantId(deviceIdInfo.getTenantId())
                 .deviceId(deviceIdInfo.getDeviceId())
                 .deviceCreationTime(getEntryValue(ed, EntityKeyType.ENTITY_FIELD, "createdTime", 0L))
                 .metaData(md)
                 .state(deviceState).build();
+        transformInactivityTimeout(deviceStateData, ed);
+        return deviceStateData;
+    }
+
+    private void transformInactivityTimeout(DeviceStateData deviceStateData, EntityData ed) {
```

Comment:
What is the purpose of extracting this to a separate method, if you use it only once?